data_handler/handlers/health_ratio_level/health_ratio_handlers.py

```diff
@@ -37,14 +42,201 @@ def fetch_data(self, protocol_name: str) -> tuple:
+        :return: State
+        """
+        for instance in data:
+            loan_entity = self.loan_entity_class()
+
+            loan_entity.debt = TokenValues(values=instance.debt)
+            loan_entity.collateral = TokenValues(values=instance.collateral)
+
+            state.loan_entities.update(
+                {
+                    instance.user: loan_entity,
+                }
+            )
+
+        return state
+
+    def calculate_health_ratio(self) -> defaultdict:
         """
         Calculates health ratio based on provided data.
         :return: A dictionary of the ready health ratio data.
         """
         data, interest_rate_models = self.fetch_data(protocol_name=ProtocolIDs.ZKLEND.value)
         state = self.state_class()
+        state = self.initialize_loan_entities(state=state, data=data)
+
+        # Set up collateral and debt interest rate models
+        state.collateral_interest_rate_models = TokenValues(
+            values=interest_rate_models.collateral
+        )
+        state.debt_interest_rate_models = TokenValues(
+            values=interest_rate_models.debt
+        )
+
+        current_prices = Prices()
+        asyncio.run(current_prices.get_lp_token_prices())
+
+        result_data = defaultdict()
+        prices = TokenValues(values=current_prices.prices.values)
+
+        for user_id, loan_entity in state.loan_entities.items():
+            risk_adjusted_collateral_usd = loan_entity.compute_collateral_usd(
+                risk_adjusted=True,
+                collateral_interest_rate_models=state.collateral_interest_rate_models,
+                prices=prices,
+            )
+            debt_usd = loan_entity.compute_debt_usd(
+                risk_adjusted=False,
+                debt_interest_rate_models=state.debt_interest_rate_models,
+                prices=prices,
+            )
+            health_ratio_level = loan_entity.compute_health_factor(
+                standardized=False,
+                risk_adjusted_collateral_usd=risk_adjusted_collateral_usd,
+                debt_usd=debt_usd,
+            )
+
+            if self.health_ratio_is_valid(health_ratio_level):
+                result_data.update({
+                    f"{uuid.uuid4()}": {
+                        USER_FIELD_NAME: user_id,
+                        HEALTH_FACTOR_FIELD_NAME: health_ratio_level,
+                        TIMESTAMP_FIELD_NAME: datetime.now().timestamp()
+                    }
+                })
+
+        return result_data
+
+
+class NostrAlphaHealthRatioHandler(BaseHealthRatioHandler):
+    """
+    A Nostra Alha handler that collects data from DB,
+    computes health_ratio level and stores it in the database.
+
+    :cvar CONNECTOR: A DB connection object.
+    """
+    def __init__(self):
+        super().__init__(state_class=NostraAlphaState, loan_entity_class=NostraAlphaLoanEntity)
+
+    def initialize_loan_entities(self, state: State, data: dict = None) -> State:
```

Comment:
I think this method can be unified in one, as far as it's using everywhere  I mean for Zklend and Both Nostra classes

Reply:
Feel free to move it to `BaseHealthRatioHandler`